server/events/policy_check_output_generator.go

```diff
@@ -0,0 +1,125 @@
+package events
+
+import (
+	"fmt"
+
+	"github.com/runatlantis/atlantis/server/events/command"
+	"github.com/runatlantis/atlantis/server/events/models"
+	"github.com/runatlantis/atlantis/server/lyft/feature"
+)
+
+type PolicyCheckOutputKey struct {
+	ProjectName string
+	Workspace   string
+}
+
+type PolicyCheckOutputStore struct {
+	Store map[PolicyCheckOutputKey]*models.PolicyCheckSuccess
+}
+
+func NewPolicyCheckOutputStore() *PolicyCheckOutputStore {
+	return &PolicyCheckOutputStore{
+		Store: map[PolicyCheckOutputKey]*models.PolicyCheckSuccess{},
+	}
+}
+
+func (p *PolicyCheckOutputStore) GetOutputFor(projectName string, workspace string) *models.PolicyCheckSuccess {
+	key := PolicyCheckOutputKey{
+		ProjectName: projectName,
+		Workspace:   workspace,
+	}
+
+	if ouptut, ok := p.Store[key]; ok {
+		return ouptut
+	}
+	return nil
+}
+
+func (p *PolicyCheckOutputStore) WriteOutputFor(projectName string, workspace string, policyCheckResult command.ProjectResult) {
+
+	key := PolicyCheckOutputKey{
+		ProjectName: projectName,
+		Workspace:   workspace,
+	}
+
+	var output string
+	if policyCheckResult.Failure != "" {
+		output = policyCheckResult.Failure
+	} else if policyCheckResult.PolicyCheckSuccess != nil {
+		output = policyCheckResult.PolicyCheckSuccess.PolicyCheckOutput
+	}
+
+	val := &models.PolicyCheckSuccess{
+		PolicyCheckOutput: output,
+	}
+
+	p.Store[key] = val
+}
+
+type CommandOutputGenerator interface {
+	GenerateCommandOutput(ctx *command.Context, cmd *command.Comment) PolicyCheckOutputStore
+}
+
+type NoopCommandOutputGenerator struct{}
+
+func (n *NoopCommandOutputGenerator) GenerateCommandOutput(ctx *command.Context, cmd *command.Comment) PolicyCheckOutputStore {
+	return PolicyCheckOutputStore{}
+}
```

Comment:
Hmm not sure, I originally had it in the same file but moved it to a different one to avoid having one single big file. @nishkrishnan what do you think?